src/config.rs

```diff
@@ -2771,6 +2778,23 @@ mod tests {
         assert!(tikv_cfg.check_critical_cfg_with(&last_cfg).is_ok());
     }
 
+    #[test]
+    fn test_last_cfg_not_modified_if_it_is_equivalent() {
```

Comment:
Should also check updates are actually triggering writes.

Reply:
Any progress on this?

Reply:
I'll add a test about this.